src/main/scala/is/hail/expr/ir/ComparisonOp.scala

```diff
@@ -0,0 +1,62 @@
+package is.hail.expr.ir
+
+import is.hail.annotations.CodeOrdering
+import is.hail.asm4s.Code
+import is.hail.expr.types._
+import is.hail.utils.lift
+
+object ComparisonOp {
```

Comment:
Can you opine a bit on why we're divorcing comparisons from binary primitive operations? We could also implement this by teaching binary primitive operations the distinction between strict- and lazy-missingness.

Reply:
because i did a dumb. I'll move them back.

Reply:
actually, the interface for emitting code was kind of hard to unify here, so i might leave as is.